File: test/app/background.dart
library background;

import 'package:chrome/app.dart';

void main() {
  runtime.onMessage.listen((MessageEvent evt) {
    evt.sendResponse('respond: ${evt.message}');
  });

  app.window.create('harness_browser.html', 
      id: 'chrome_test', minWidth: 1024, minHeight: 768);
}

File: app/test_app/background.js

chrome.runtime.onMessage.addListener(function(message, sender, sendResponse) {
  console.log('respond: ' + message);
  sendResponse('respond: ' + message);
});

chrome.app.runtime.onLaunched.addListener(function() {
  chrome.app.window.create('harness.html',
    {id: 'chrome_test', width: 1024, height: 768});
});
